src/lib/grammar.rs

```diff
@@ -224,10 +224,20 @@ impl Grammar {
         self.prod_precs.get(usize::from(i)).map_or(None, |x| Some(*x))
     }
 
+    /// Return the name of terminal `i` or None if it doesn't exist.
+    pub fn term_name(&self, i: TIdx) -> Option<&str> {
+        self.terminal_names.get(usize::from(i)).map_or(None, |x| Some(&x))
+    }
+
     /// Return the precedence of terminal `i` or None if it doesn't exist.
     pub fn term_precedence(&self, i: TIdx) -> Option<Option<Precedence>> {
         self.terminal_precs.get(usize::from(i)).map_or(None, |x| Some(*x))
     }
+
+    /// Return an iterator which produces (in no particular order) all this grammar's valid TIdxs.
+    pub fn iter_term_idxs(&self) -> Box<Iterator<Item=TIdx>> {
```

Comment:
This function is not used in this commit. I assume it's being used later in the PR?